controllers/disruption_controller.go

```diff
@@ -600,32 +600,27 @@ func (r *DisruptionReconciler) selectTargets(instance *chaosv1beta1.Disruption)
 		}
 	}
 
-	// prune ignored targets from all targets
-	for _, target := range matchingTargets {
-		if !contains(instance.Status.IgnoredTargets, target) {
-			eligibleTargets = append(eligibleTargets, target)
-		}
-	}
-
 	// return an error if the selector returned no targets
-	if len(eligibleTargets) == 0 {
+	if len(matchingTargets) == 0 {
 		r.log.Info("the given label selector did not return any targets, skipping")
 		r.Recorder.Event(instance, "Warning", "NoTarget", "The given label selector did not return any targets. Please ensure that both the selector and the count are correct (should be either a percentage or an integer greater than 0).")
 
 		return nil
 	}
 
+	// prune ignored targets from all targets
```

Comment:
I would even move this below `getScaledValueFromIntOrPercent`, and my personal preference would be to declare `eligibleTargets := []string{}` right above this for loop as well, though I believe that idea was rejected a few months ago, don't remember why

Reply:
Makes sense!